cmake/kokkos_arch.cmake

```diff
@@ -744,14 +744,32 @@ ENDIF()
 
 IF (KOKKOS_ENABLE_OPENACC)
   IF(KOKKOS_CUDA_ARCH_FLAG)
+    SET(CLANG_CUDA_ARCH ${KOKKOS_CUDA_ARCH_FLAG})
     STRING(REPLACE "sm_" "cc" NVHPC_CUDA_ARCH ${KOKKOS_CUDA_ARCH_FLAG})
     COMPILER_SPECIFIC_FLAGS(
       NVHPC -acc -gpu=${NVHPC_CUDA_ARCH}
+      Clang -Xopenmp-target=nvptx64-nvidia-cuda -march=${CLANG_CUDA_ARCH}
+            -fopenmp-targets=nvptx64-nvidia-cuda
+    )
+  ELSEIF(KOKKOS_AMDGPU_ARCH_FLAG)
+    COMPILER_SPECIFIC_FLAGS(
+      Clang -Xopenmp-target=amdgcn-amd-amdhsa -march=${KOKKOS_AMDGPU_ARCH_FLAG}
+            -fopenmp-targets=amdgcn-amd-amdhsa
+    )
+    COMPILER_SPECIFIC_LIBS(
+      Clang -lm
     )
   ELSE()
     COMPILER_SPECIFIC_FLAGS(
       NVHPC -acc
     )
+    # When not compiling for offload to any GPU, we're compiling for kernel
+    # execution on the host.  In that case, memory is shared between the OpenACC
+    # space and the host space.
+    COMPILER_SPECIFIC_DEFS(
+      Clang KOKKOS_OPENACC_WITHOUT_GPU
+      NVHPC KOKKOS_OPENACC_WITHOUT_GPU
+    )
```

Comment:
Do we actually care about that case apart from debugging (and then the shared memory space doesn't really help)?

Reply:
A user may want to execute a kernel on a host (in parallel).